tests/conftest.py

```diff
@@ -59,3 +69,201 @@ def memory_keyring():
     keyring.set_keyring(MemoryKeyring())
     yield
     keyring.set_keyring(current_keyring)
+
+
+class Platform(models.CraftBaseModel):
+    """Platform definition."""
+
+    build_on: str
+    build_for: str
+
+
+class MyBuildPlanner(models.BuildPlanner):
+    """Build planner definition for tests."""
+
+    def get_build_plan(self) -> list[models.BuildInfo]:
+        arch = util.get_host_architecture()
+        return [
+            models.BuildInfo(
+                "ubuntu-22.04", arch, arch, bases.BaseName("ubuntu", "22.04")
```

Comment:
I'm not sure why these test fixtures are being copied from craft-application, but I think these bases should specify `24.04` at least